src/Components/SingleArticle.jsx

```diff
@@ -0,0 +1,102 @@
+import { useEffect, useState } from "react"
+import { getSingleArticle } from "../api/api.js"
+import { patchVoteCount } from "../api/api.js"
+
+import Comments from './Comments.jsx'
+
+import {useParams} from 'react-router-dom';
+
+export default function SingleArticle(){
+
+    const {article_id} = useParams();    
+    
+    const [article, setArticle] = useState({});
+    const [localVotes, setLocalVotes] = useState(0);
+
+    const [isLoading, setIsLoading] = useState(true);
+    const [isError, setIsError] = useState(null);
+    const [isVoteError, setIsVoteError] = useState(false);
+    const [hasUpvoted, setHasUpvoted] = useState(false);
+    const [hasDownvoted, setHasDownvoted] = useState(false);
+    const [errorMsg, setErrorMsg] = useState(null);
+
+    useEffect(()=>{
+        setIsLoading(true);
+        getSingleArticle(article_id).then((article) => {
+            setIsLoading(false);
+            setIsError(false);
+            setArticle(article);
+            setLocalVotes(article.votes)
+        })
+        .catch((err)=>{
+            setIsError(true)
+            setErrorMsg(err.message)
+        })
+    }, [article_id])
+
+    function handleVote(id, vote){
+        {setLocalVotes(localVotes + vote)}
```

Comment:
when using the previous state in the calculation of the new state you should opt for the callback version of the function